### DISCUSSION TITLE
Module vs Script

### DISCUSSION THREAD

**Discussion by ghost** [HotCakeX/Harden-Windows-Security#107] [Q&A]
Hi! What is the difference between the windows security module and the windows security script?

Thank You!

**Comment by HotCakeX** [comment]
Hi,
The module is newer, has more features (Applying hardening measures, Removing hardening measures, Checking/Auditing hardening measures), it requires new PowerShell core. (Installable using Winget, GitHub or Microsoft store)

The [script](https://www.powershellgallery.com/packages/Harden-Windows-Security) is for users that use old PowerShell 5.1, the one that comes with Windows. It has less features.

**Comment by ghost** [comment]
Oh great! So with that being said. If I have a new machine and from the very start should I use the module after I install the new PowerShell from the Windows store?

**Reply by HotCakeX** [reply]
Yes that's correct

**Reply by ghost** [reply]
Hi! If I wanted to apply the new update to the module do I need to run the install command every time?

**Reply by HotCakeX** [reply]
Hi, no, just run the module normally using any of its cmdlets.

**Reply by ghost** [reply]
Perfect so I can just do the “protect windows security” and be good?

**Reply by HotCakeX** [reply]
Yes
